python/cudf/cudf/core/column/column.py

```diff
@@ -2184,11 +2240,13 @@ def as_column(
                 arbitrary = arbitrary.astype(cudf.dtype("datetime64[s]"))
 
             buffer = as_buffer(arbitrary.view("|u1"))
-            mask = None
             if nan_as_null is None or nan_as_null is True:
                 data = build_column(buffer, dtype=arbitrary.dtype)
                 data = _make_copy_replacing_NaT_with_null(data)
                 mask = data.mask
+            else:
+                bool_mask = as_column(~np.isnat(arbitrary))
+                mask = as_buffer(bools_to_mask(bool_mask))
```

Comment:
This seems like it turns `NaT` into nulls, is that the intention?

Reply:
Yeah I think we still want to consider `NaT` values when creating the mask (as null values), but not necessarily cast the value to NA as tested in `test_series_np_array_nat_nan_as_null_false`. cc @galipremsagar

Reply:
Yeah, at one point of time we supported both `NA` and `NaT` for datetime & timedelta columns. Now we just treat `NA` as `NaT` and vice-versa for datetime & timedelta column. With recent pandas accelerator mode work, we decided to just repr out `NA` as `NAT`. So yes we need to mark the mask when we have `NAT`'s anywhere.